udf-compiler/src/test/scala/com/nvidia/spark/OpcodeSuite.scala

```diff
@@ -45,19 +42,1834 @@ class OpcodeSuite extends FunSuite {
 
   import spark.implicits._
 
-// Utility Function for checking equivalency of Dataset type
+  // Utility Function for checking equivalency of Dataset type
   def checkEquiv[T](ds1: Dataset[T], ds2: Dataset[T]) : Unit = {
-    ds1.explain(true)
-    ds2.explain(true)
     val resultdf = ds1.toDF()
     val refdf = ds2.toDF()
-    ds1.show
-    ds2.show
     val columns = refdf.schema.fields.map(_.name)
     val selectiveDifferences = columns.map(col => refdf.select(col).except(resultdf.select(col)))
     selectiveDifferences.map(diff => { assert(diff.count==0) } )
-    ds1.show
-    ds2.show
   }
-}
 
+  // START OF TESTS
+
+  test("not boolean") {
+    val myudf: Boolean => Boolean = { x => !x }
+    val u = makeUdf(myudf)
+    val dataset = List(true, false, true, false).toDS().repartition(1)
+    val result = dataset.withColumn("new", u('value))
+    val equiv = dataset.withColumn("new", not(col("value")))
+    checkEquiv(result, equiv)
+  }
+
+  test("and boolean") {
+    val myudf: (Boolean, Boolean) => Boolean = { (a, b) => a && b }
+    val u = makeUdf(myudf)
+    val df = List((false, false), (false, true), (true, false), (true, true))
+      .toDF("x", "y")
+      .repartition(1)
+    val result = df.withColumn("new", u(col("x"), col("y")))
+    val equiv = df.withColumn("new", col("x") && col("y"))
+    checkEquiv(result, equiv)
+  }
+
+
+  test("not(and boolean)") {
+    val myudf: (Boolean, Boolean) => Boolean = { (a, b) => !(a && b) }
+    val u = makeUdf(myudf)
+    val df = List((false, false), (false, true), (true, false), (true, true))
+      .toDF("x", "y")
+      .repartition(1)
+    val result = df.withColumn("new", u(col("x"), col("y")))
+    val equiv = df.withColumn("new", col("x") && col("y"))
+    checkEquiv(result, equiv)
+  }
+
+  test("or boolean") {
+    val myudf: (Boolean, Boolean) => Boolean = { (a, b) => a || b }
+    val u = makeUdf(myudf)
+    val df = List((false, false), (false, true), (true, false), (true, true))
+      .toDF("x", "y")
+      .repartition(1)
+    val result = df.withColumn("new", u(col("x"), col("y")))
+    val equiv = df.withColumn("new", col("x") || col("y"))
+    checkEquiv(result, equiv)
+  }
+
```

Comment:
This is a nit.. but it would be nice if the spacing between the tests is always 1 line.

Reply:
Cleaned.